lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/summary/IncrementalSummarizer.java

```diff
@@ -41,26 +41,35 @@
     private HashMap<Set<Integer>, Double> outlierItemsetWindowCount = new HashMap<>();
     private Deque<Integer> inlierPaneCounts;
     private Deque<Integer> outlierPaneCounts;
+    private HashMap<Set<Integer>, Integer> trackingMap = new HashMap<>();
+
+    // Temp Intermediate Values
     private List<Integer> inlierCountCumSum;
     private List<Integer> outlierCountCumSum;
-    private HashMap<Set<Integer>, Integer> trackingMap = new HashMap<>();
 
     public IncrementalSummarizer(int numPanes) {
         setWindowSize(numPanes);
-        if (inlierItemsetPaneCounts == null) {
-            inlierPaneCounts = new ArrayDeque<>(numPanes);
-            outlierPaneCounts = new ArrayDeque<>(numPanes);
-            inlierItemsetPaneCounts = new ArrayDeque<>(numPanes);
-            outlierItemsetPaneCounts = new ArrayDeque<>(numPanes);
-        }
     }
 
+    public IncrementalSummarizer() {
+        setWindowSize(1);
+    }
+
+
     @Override
     public void setWindowSize(int numPanes) {
         this.numPanes = numPanes;
+        initializePanes();
```

Comment:
We only want to call this function if `inlierPaneCounts, outlierPaneCounts` etc. are not initialized.